plugin/code_actions.py

```diff
@@ -273,25 +273,34 @@ class LspCodeActionsCommand(LspTextCommand):
 
     capability = 'codeActionProvider'
 
-    def run(self, edit: sublime.Edit, event: Optional[dict] = None, only_kinds: Optional[List[str]] = None) -> None:
+    def run(self, edit: sublime.Edit, event: Optional[dict] = None,
+            only_kinds: Optional[List[str]] = None, commands_by_config: CodeActionsByConfigName = None) -> None:
         self.commands = []  # type: List[Tuple[str, str, CodeActionOrCommand]]
         self.commands_by_config = {}  # type: CodeActionsByConfigName
-        view = self.view
-        region = first_selection_region(view)
-        if region is None:
-            return
-        listener = windows.listener_for_view(view)
-        if not listener:
-            return
-        session_buffer_diagnostics, covering = listener.diagnostics_intersecting_async(region)
-        dict_kinds = {kind: True for kind in only_kinds} if only_kinds else None
-        actions_manager.request_for_region_async(
-            view, covering, session_buffer_diagnostics, self.handle_responses_async, dict_kinds)
+        self.has_commands_by_config = False
+        if commands_by_config:
+            self.has_commands_by_config = True
```

Comment:
I'd prefer to have this flag be an optional argument of `handle_responses_async` instead so that there is no state to keep track of and less code.

Reply:
> I'd prefer to have this flag be an optional argument of `handle_responses_async` instead so that there is no state to keep track of and less code.

Resolve in 6f96859